app/models/exchange_rates/period_list.rb

```diff
@@ -9,6 +9,8 @@ class ExchangeRates::PeriodList
   has_many :exchange_rate_periods, class_name: 'ExchangeRates::Period'
 
   def publication_date
+    raise Faraday::ResourceNotFound if exchange_rate_periods.empty?
```

Comment:
Oh, now that you use this exception I'm wondering if this is actually automatically propagated given what you've plumbed together in the backend?

Reply:
I was just using similar to what we have elsewhere but i couldnt find anything else.

Reply:
Yeah, I think that Jeremy added middleware propagation of 404s https://github.com/trade-tariff/trade-tariff-frontend/blob/main/app/services/client_builder.rb#L52

You actually ought to be able to just remove this line with your backend change

Reply:
There's actually generic handling of Faraday::ResourceNotFound https://github.com/trade-tariff/trade-tariff-frontend/blob/main/config/initializers/exceptions.rb#L2

Reply:
Not saying we should use the generic view, etc, since your 404 is more informative

Reply:
Yeah, custom message is great. Just a by-the-by thing

Reply:
So basically, I think we can remove this propagation, here, and let the rescue_from/client do the work

Reply:
I tested that but didnt work for me 😶‍🌫️

Reply:
Even though no results are returned from the API with 404 i still need to check here to stop it blowing up

Reply:
Oh, that's strange. I guess the middleware isn't propagating 404s :confused: